### PR TITLE
Warning message if document as been updated while editing it

### PR DESCRIPTION
**Pull request by jenow**
Show a message when editing a document if someone else edited it at the same time.

### CODE REVIEW

File: src/components/Data/Documents/Update.vue [hunk 1]
```diff
@@ -83,17 +105,17 @@
...
         .dataCollectionFactory(this.collection, this.index)
-        .fetchDocumentPromise(this.documentToEditId)
-        .then(res => {
-          this.setNewDocument(res.content)
-          this.$broadcast('document-create::fill', res.content)
-          return null
-        })
-        .catch(err => {
-          this.$dispatch('toast', err.message, 'error')
+        .subscribe({term: {_id: this.documentToEditId}}, () => {
```
Comment (AnthonySendra) [inline src/components/Data/Documents/Update.vue]: You probably need to store the subscription and perform `room.unsubscribe()` on `beforeDestroy()`hook.

```
let subscription
[...]
export default {
[...]
ready() {
  this.fetch()
  subscription = kuzzle
    .dataCollectionFactory(this.collection, this.index)
    .subscribe([...])
},
beforeDestroy () {
  subscription.unsubscribe()
}
```

Comment (jenow) [inline src/components/Data/Documents/Update.vue]: I can't store a room in the store, it is too complex but yes I will unsubscribe on beforeDestroy.

Comment (AnthonySendra) [inline src/components/Data/Documents/Update.vue]: Didn't talk about the "application store". I just use the verb "store" :D

**Comment by AnthonySendra** [comment]
+1

**Comment by stafyniaksacha** [comment]
+1

**Pull request merged**